### PR TITLE
Migrate to cw-assets v3 and cw-storage-plus v1.2.0

### PR DESCRIPTION
**Pull request by kerber0x**
This PR upgrades the cw-assets and cw-storage-plus dependencies of the alliance-hub contract in preparation to tweaks for eris' ve3-contracts.

### CODE REVIEW

**Review by 0xFable — APPROVED** [review]
Looking pretty good. I have some concerns of not using AssetKey anymore but maybe that is a requirement for moving to cw-assets v3

File: contracts/alliance-hub/src/contract.rs [hunk 1]
```diff
@@ -262,16 +263,15 @@ fn stake(
     amount: Uint128,
     sender: Addr,
 ) -> Result<Response, ContractError> {
-    let asset_key = AssetInfoKey::from(&asset);
     WHITELIST
-        .load(deps.storage, asset_key.clone())
+        .load(deps.storage, &asset)
```
[the remark this answers is not in the document]
Comment (kerber0x) [reply contracts/alliance-hub/src/contract.rs:267]: `AssetInfoKey` does not exist in cw-assets v3, that's the whole point of this migration, should probably have mentioned it above

File: contracts/alliance-hub/src/migrations.rs [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    migrate_asset_reward_distribution(deps.branch())?;
+    migrate_asset_reward_rate(deps.branch())?;
+    migrate_user_asset_reward_rate(deps.branch())?;
+    migrate_unclaimed_rewards(deps.branch())?;
+
+    Ok(())
+}
+
+fn migrate_whitelist_map(deps: DepsMut) -> Result<(), ContractError> {
+    const OLD_WHITELIST: Map016<cw_asset_v2::AssetInfoKey, ChainId> = Map016::new("whitelist");
```
[the remark this answers is not in the document]
Comment (kerber0x) [reply contracts/alliance-hub/src/migrations.rs:25]: correct

File: contracts/alliance-hub/src/contract.rs [hunk 3]
```diff
@@ -603,22 +591,21 @@ fn update_reward_callback(
...
     for asset_distribution in asset_reward_distribution {
-        let asset_key = AssetInfoKey::from(asset_distribution.asset);
         let total_reward_distributed = Decimal::from_atomics(rewards_collected, 0)?
             * asset_distribution.distribution
             / total_distribution;
 
         // If there are no balances, we stop updating the rate. This means that the emissions are not directed to any stakers.
         let total_balance = TOTAL_BALANCES
-            .load(deps.storage, asset_key.clone())
+            .load(deps.storage, &asset_distribution.asset)
```
[the remark this answers is not in the document]
Comment (kerber0x) [reply contracts/alliance-hub/src/contract.rs:600]: all of it causes migrations 😅

**Pull request merged**